File: problems/0003-longest-substring-without-repeating-characters/solution.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        l =0
        longest= 0
        st = set()
        for i in range(len(s)):
            while s[i] in st:
                st.remove(s[l])
                l+=1
            st.add(s[i])
            longest = max(longest,i-l+1)
        return longest    




        

File: problems/0003-longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max = 0;
        for (int i = 0; i < s.length(); i++) {
            String current = "";
            for (int j = i; j < s.length(); j++) {
                char c = s.charAt(j);
                if (current.indexOf(c) != -1) {
                    break;
                } else {
                    current += c; 
                }
            }
            if(max<current.length()){
                max = current.length();
            }
        }
        return max;
    }
}
